libraries/ui-library-angular/src/lib/form/six-form.directive.ts

```diff
@@ -0,0 +1,111 @@
+ * takes proactive actions by marking all form controls as touched and dirty. Additionally,
+ * it shifts the focus to the initial invalid form element, facilitating quick error
+ * resolution.
+ *
+ * To utilize this directive, apply it to an Angular form.
+ * ```html
+ * <form [formGroup]="form" sixForm (sixSubmit)="onSubmit($event)">
+ *   <!-- form content -->
+ * </form>
+ * ```
+ *
+ * For users needing greater flexibility in determining when error messages are displayed,
+ * or for those who prefer not to rely solely on the form submission event,
+ * an alternative is to use the SixFormUtilDirective.
+ */
+@Directive({
+  selector: 'form[sixForm]',
+})
+export class SixFormDirective {
+  /**
+   * Emits an event when the form is valid and the form submission has been triggered.
+   */
+  @Output() sixSubmit = new EventEmitter<SubmitEvent>();
+
+  @HostListener('ngSubmit', ['$event'])
+  onNgSubmit(event: SubmitEvent): void {
+    const formGroupDirective = this.injector.get(FormGroupDirective);
+    if (formGroupDirective.invalid) {
+      focusInvalidField(formGroupDirective, this.elementRef);
+    } else {
+      this.sixSubmit.emit(event);
+    }
+  }
+
+  constructor(private elementRef: ElementRef<HTMLElement>, private injector: Injector) {}
+}
+
+/**
+ * This directive provides a utility method, that marks all form controls
+ * as touched and dirty, and focuses the first invalid form element.
+ *
+ * To utilize this directive, apply it to an Angular form.
+ * ```html
+ * <form [formGroup]="form" sixFormUtil (ngSubmit)="onSubmit($event)">
+ *   <!-- form content -->
+ * </form>
+ * ```
+ *
+ * Then, get a reference to the directive and invoke `focusInvalidField()` if the
+ * form is invalid:
+ * ```ts
+ * @ViewChild(SixFormUtilDirective) sixFormUtil!: SixFormUtilDirective;
+ * // ...
+ * onSubmit() {
+ *   if (this.form.invalid) {
+ *     this.sixFormUtil.focusInvalidField();
+ *   } else {
+ *      // ...
+ *   }
+ * }
+ * ```
+ */
+@Directive({
+  selector: '[sixFormUtil]',
+})
+export class SixFormUtilDirective {
+  constructor(private elementRef: ElementRef<HTMLElement>, private injector: Injector) {}
+
+  /**
+   * Marks all form controls as touched and dirty, and focuses the first
+   * invalid form element.
+   */
+  public focusInvalidField() {
+    focusInvalidField(this.injector.get(FormGroupDirective), this.elementRef);
+  }
+}
+
+function focusInvalidField(formGroupDirective: FormGroupDirective, formElement: ElementRef<HTMLElement>) {
+  if (formGroupDirective) {
+    formGroupDirective.form.markAllAsTouched();
+    markAllAsDirty(formGroupDirective.form);
+    const invalidField: any = formElement.nativeElement.querySelector('.ng-invalid');
+    if (typeof invalidField?.setFocus === 'function') {
+      invalidField.setFocus();
+    } else if (typeof invalidField?.focus === 'function') {
+      invalidField?.focus();
+    }
+  }
+}
+
+function markAllAsDirty(formGroup: FormGroup) {
```

Comment:
Are you intentionally not marking FormGroup / FormArray as dirty?
Since onlySelf is used for FormControls this might be a problem in case of validators on FormArray / FormGroup.

Reply:
Indeed, form group is not set to `dirty` with the current logic. Fixed it.